recordedit/recordEdit.app.js

```diff
@@ -183,6 +183,15 @@
                 } else {
                     if ($rootScope.reference.canCreate) {
                         $rootScope.displayname = $rootScope.reference.displayname;
+
+                        // populate defaults
+                        angular.forEach($rootScope.reference.columns, function(column) {
+                            // typeof column.default !== "function"
+                            if (column.default) {
```

Comment:
Thinking out loud: In the scenario when the the URL has a `prefill` cookie but no filter, then the app will go to `if (context.prefill)` to fill in the foreign key link's view model, and then skip `if (context.filter)` to go to its else branch and apply this `angular.forEach` loop.

Is there a chance of this loop overwriting the prefill's values? (And if so, is that a concern anyway because does it make sense for foreign key columns to have defaults?)